framework/doc/content/source/convergence/PostprocessorConvergence.md

```diff
@@ -7,7 +7,7 @@ and compares a [Postprocessor](Postprocessors/index.md) value $y$ to a tolerance
 |y| \leq \tau \,.
 
 For this to work as expected, the `execute_on` parameter of the post-processor
-must include values that trigger execution before the desired check. For example, for assessing convergence of the  nonlinear solve, the value `NONLINEAR_CONVERGENCE` should be used.
+must include values that trigger execution before the desired check. For example, for assessing convergence of the  nonlinear solve, the value `NONLINEAR_CONVERGENCE` should be used. For assessing convergence of a MultiApp fixed point solve, the appropriate `execute_on` depends on when the MultiApps are executed and on the post-processor type. For example, for [/Residual.md], `TIMESTEP_BEGIN` is appropriate for MultiApps executing on `TIMESTEP_BEGIN`, and `MULTIAPP_FIXED_POINT_CONVERGENCE` is appropriate for MultiApps executing on `TIMESTEP_END`. See [SetupInterface.md] for details on different execution points.
```

Comment:
are you sure that the convergence check on multiapps is not always at timestep_end (really at `MULTIAPP_FIXED_POINT_CONVERGENCE ?) 

and in that case I think it does not need to be executed on timestep_begin.

Especially if it is based on the values in the parent application, I would expect timestep_end (or multiapp_fp_convergence?) to still make more sense

Reply:
> I have verified that what I listed for Residual is true, and for TIMESTEP_BEGIN MultiApps, executing Residual on TIMESTEP_BEGIN produces the desired result

This does not seem right. That would be before the nonlinear solve? Maybe you are getting the residual from the previous time step, just with the updated fields coming from the multiapps?

We are talking about the parent app residual here right?

> If there is no transfer after the end of the parent's solve, then the residual you get if you execute Residual at TIMESTEP_END is the same as the converged residual at the end of the parent's nonlinear solve (very small).

Agree with that. timestep_end is right after the parent's nonlinear solve

Reply:
> This does not seem right. That would be before the nonlinear solve? Maybe you are getting the residual from the previous time step, just with the updated fields coming from the multiapps?

discussed on slack. Basically once we solved on timestep_end we have a tiny residual regardless of the effect of the multiapps on timestep_begin. So timestep_begin is appropriate.